src/core/gps/qgsgpsconnection.cpp

```diff
@@ -25,6 +25,48 @@
 
 #include "qgsnmeaconnection.h"
 #include "qgslogger.h"
+#include "info.h"
+
+
+bool QgsGpsInformation::isValid() const
+{
+  bool valid = false;
+  if ( status == 'V' || fixType == NMEA_FIX_BAD || quality == 0 ) // some sources say that 'V' indicates position fix, but is below acceptable quality
+  {
+    return false;
```

Comment:
Can we either have a local return value variable and a single return statement in the end or directly return the value everywhere?
The current code is very confusing to read.

```suggestion
    valid = false;
```

Reply:
Fixed